crates/pallet-domains/src/staking.rs

```diff
@@ -497,19 +518,60 @@ pub(crate) fn do_convert_previous_epoch_deposits<T: Config>(
     Ok(())
 }
 
+pub(crate) fn allowed_default_share_price<T: Config>(
+    operator_id: OperatorId,
+    domain_epoch_index: EpochIndex,
+) -> Option<SharePrice> {
+    let DomainEpoch(allowed_domain_id, allowed_epoch_index) =
+        AllowedDefaultSharePriceEpoch::<T>::get()?;
+    let operator = Operators::<T>::get(operator_id)?;
+
+    // The default share price is only allowed if the requested `domain_epoch_index` is happen
+    // before (or is the same as) the `AllowedDefaultSharePriceEpoch`
+    if allowed_domain_id == operator.current_domain_id && allowed_epoch_index >= domain_epoch_index
+    {
+        let domain_stake_summary = DomainStakingSummary::<T>::get(operator.current_domain_id)?;
+        return Some(current_share_price::<T>(
+            operator_id,
+            &operator,
+            &domain_stake_summary,
+        ));
+    }
+    None
+}
+
 /// Converts any epoch withdrawals into balance using the operator epoch share price.
 /// If there is no share price available, this will be no-op
```

Comment:
This documentation might need to be updated to describe the default share price and MissingOperatorEpochSharePrice error cases.